nkms/network/server.py

```diff
@@ -248,25 +253,31 @@ def reencrypt_via_rest(self, hrac_as_hex, request: http.Request):
     def provide_treasure_map(self, treasure_map_id_as_hex):
         # For now, grab the TreasureMap for the DHT storage.  Soon, no do that.  #TODO!
         treasure_map_id = binascii.unhexlify(treasure_map_id_as_hex)
-        treasure_map_bytes = self.server.storage.get(digest(treasure_map_id))
         headers = {'Content-Type': 'application/octet-stream'}
 
-        return Response(content=treasure_map_bytes, headers=headers)
+        try:
+            treasure_map_bytes = self.server.storage[digest(treasure_map_id)]
+            response = Response(content=treasure_map_bytes, headers=headers)
+        except KeyError:
+            response = Response("No Treasure Map with ID {}".format(treasure_map_id),
+                                status_code=404, headers=headers)
+
+        return response
 
     def receive_treasure_map(self, treasure_map_id_as_hex, request: http.Request):
         # TODO: This function is the epitome of #172.
         treasure_map_id = binascii.unhexlify(treasure_map_id_as_hex)
 
         header, signature_for_ursula, pubkey_sig_alice, hrac, tmap_message_kit = \
-            dht_value_splitter(request.body, return_remainder=True)
+            dht_with_hrac_splitter(request.body, return_remainder=True)
         # TODO: This next line is possibly the worst in the entire codebase at the moment.  #172.
         # Also TODO: TTL?
         do_store = self.server.protocol.determine_legality_of_dht_key(
                     signature_for_ursula, pubkey_sig_alice, tmap_message_kit,
                     hrac, digest(treasure_map_id), request.body)
         if do_store:
-            # TODO: Stop storing things in the protocol storage.  Do this better.
-            # TODO: Propagate to other nodes.
+            # TODO: Stop storing things in the protocol storage.  Do this better.  #227
+            # TODO: Propagate to other nodes.  #235
```

Comment:
Great work in enumerating all of these issues. Many thanks. :pray:

Reply:
Still many left to do!  :-)

I encourage you and @KPrasch to do the same as you're putting together #237.